dartsim/src/JointFeatures_TEST.cc

```diff
@@ -1056,6 +1064,334 @@ TEST_F(JointFeaturesFixture, JointAttachDetachSpawnedModel)
+    ASSERT_NE(nullptr, this->model);
+    this->motorJoint = this->model->GetJoint("motor_joint");
+    ASSERT_NE(nullptr, this->motorJoint);
+    this->sensorJoint = this->model->GetJoint("sensor_joint");
+    ASSERT_NE(nullptr, this->sensorJoint);
+    this->armLink = this->model->GetLink("arm");
+    ASSERT_NE(nullptr, this->armLink);
+  }
+
+  public: void Step(int _iters)
+  {
+    for (int i = 0; i < _iters; ++i)
+    {
+      this->world->Step(this->output, this->state, this->input);
+    }
+  }
+
+  public: physics::ForwardStep::Output output;
+  public: physics::ForwardStep::State state;
+  public: physics::ForwardStep::Input input;
+  public: WorldPtr world;
+  public: ModelPtr model;
+  public: JointPtr motorJoint;
+  public: JointPtr sensorJoint;
+  public: LinkPtr armLink;
+
+  // From SDFormat file
+  static constexpr double kGravity = 9.8;
+  static constexpr double kArmLinkMass = 6.0;
+  static constexpr double kSensorLinkMass = 0.4;
+  // MOI in the z-axis
+  static constexpr double kSensorLinkMOI = 0.02;
+  static constexpr double kArmLength = 1.0;
+};
+
+TEST_F(JointTransmittedWrenchFixture , PendulumAtZeroAngle)
+{
+  namespace test = physics::test;
+
+  // Run a few steps for the constraint forces to stabilize
+  this->Step(10);
+
+  // Test wrench expressed in different frames
+  {
+    auto wrenchAtMotorJoint = this->motorJoint->GetTransmittedWrench();
+    Wrench3d expectedWrenchAtMotorJoint{
+        Vector3d::Zero(), {-kGravity * (kArmLinkMass + kSensorLinkMass), 0, 0}};
+
+    EXPECT_TRUE(
+        test::Equal(expectedWrenchAtMotorJoint, wrenchAtMotorJoint, 1e-4));
+  }
+  {
+    auto wrenchAtMotorJointInWorld = this->motorJoint->GetTransmittedWrench(
+        this->motorJoint->GetFrameID(), physics::FrameID::World());
+    Wrench3d expectedWrenchAtMotorJointInWorld{
+        Vector3d::Zero(), {0, 0, kGravity * (kArmLinkMass + kSensorLinkMass)}};
+
+    EXPECT_TRUE(test::Equal(expectedWrenchAtMotorJointInWorld,
+                            wrenchAtMotorJointInWorld, 1e-4));
+  }
+  {
+    auto wrenchAtMotorJointInArm = this->motorJoint->GetTransmittedWrench(
+        this->armLink->GetFrameID(), this->armLink->GetFrameID());
+    // The arm frame is rotated by 90° in the Y-axis of the joint frame.
+    Wrench3d expectedWrenchAtMotorJointInArm{
+        Vector3d::Zero(), {0, 0, kGravity * (kArmLinkMass + kSensorLinkMass)}};
+
+    EXPECT_TRUE(test::Equal(expectedWrenchAtMotorJointInArm,
+                            wrenchAtMotorJointInArm, 1e-4));
+  }
+}
+
+TEST_F(JointTransmittedWrenchFixture, PendulumInMotion)
+{
+  namespace test = physics::test;
+  // Start pendulum at 90° (parallel to the ground) and stop when its around 40°
+  // so that we have non-trivial test expectations.
+  this->motorJoint->SetPosition(0, IGN_DTOR(90.0));
+  this->Step(350);
+
+  // Given the position (θ), velocity (ω), and acceleration (α) of the joint
+  // and distance from the joint to the COM (r), we can the reaction forces in
```

Comment:
nit: missing a verb in "we can the reaction forces"

compute? expect?

or just remove "we can"?